Lib/zipfile/__init__.py

```diff
@@ -563,6 +565,15 @@ def _decodeExtra(self, filename_crc):
                 except UnicodeDecodeError as e:
                     raise BadZipFile('Corrupt unicode path extra field (0x7075): invalid utf-8 bytes') from e
 
+            elif tp == 0x5455:
+                if ln == 0:
+                    raise BadZipFile('Empty extended timestamp extra field (0x5455)') from None
```

Comment:
We can drop the `from None` when not raising from within an except block.

```suggestion
                    raise BadZipFile('Empty extended timestamp extra field (0x5455)')
```